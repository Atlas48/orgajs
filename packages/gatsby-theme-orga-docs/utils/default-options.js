const path = require('path')

const defaultOptions = {
  location: ['docs'],
  imageMaxWidth: 1380,
  components: {},
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
