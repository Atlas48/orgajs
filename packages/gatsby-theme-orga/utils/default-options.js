const path = require('path')

const defaultOptions = {
  contentPath: 'content',
  filter: () => true,
  pagination: 0,
  columns: 2,
  indexPath: '/',
  imageMaxWidth: 1380,
  categoryIndexPath: category => `/${category}`,
  tagIndexPath: tag => `/:${tag}:`,
  postPath: ({ category, export_file_name }) =>
    path.resolve('/', category || '', export_file_name || ''),
}

module.exports = options => {
  return {
    ...defaultOptions,
    ...options,
  }
}
