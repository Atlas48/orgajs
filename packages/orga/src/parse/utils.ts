import { Node } from 'unist'
import { push } from '../node'
import { Lexer } from '../tokenize'
import { Parent, Token } from '../types'

export default (lexer: Lexer) => {
  const { peek, eat, save, restore } = lexer
  const collect = (stop: (n: Token) => boolean) => (container: Parent): Parent => {
    const token = peek()
    if (!token || stop(token)) return container
    eat()
    push(container)(token)
    return collect(stop)(container)
  }

  const skip = (predicate: (token: Token) => boolean): void => {
    const token = peek()
    if (token && predicate(token)) {
      eat()
      skip(predicate)
      return
    }
    return
  }

  const tryTo = <T extends Node>(
    parse: (lexer: Lexer) => T | undefined,
    action: (node: T) => void,
  ): boolean => {
    const savePoint = save()
    const node = parse(lexer)
    if (!node) {
      restore(savePoint)
      return false
    }
    action(node)
    return true
  }

  return {
    collect,
    skip,
    tryTo,
  }
}
