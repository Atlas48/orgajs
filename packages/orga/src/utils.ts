import { Token } from './types'
import { PhrasingContent } from 'orga';

const matchOperatorsRe = /[|\\{}()[\]^$+*?.]/g

export const escape = (str: string): string => {
  return str.replace(matchOperatorsRe, '\\$&')
}

export const isPhrasingContent = (token: Token): token is PhrasingContent => {
  return token.type.startsWith('text.')
    || token.type === 'footnote.reference'
    || token.type === 'link'
}
