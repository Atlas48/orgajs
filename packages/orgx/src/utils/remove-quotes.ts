export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')

export default removeQuotes
