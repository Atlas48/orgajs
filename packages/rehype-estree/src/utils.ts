export const removeQuotes = (text: string) =>
  text.trim().replace(/^["'](.+(?=["']$))["']$/, '$1')
