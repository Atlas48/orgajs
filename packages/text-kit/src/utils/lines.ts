import { Point, Position } from 'unist'
import { CoreAPI } from '../core'

export default <T extends CoreAPI>(core: T) => {
  const linePosition = (
    start: number | Point,
    end?: number | Point
  ): Position | undefined => {
    let result: Position | undefined

    if (typeof start === 'number') {
      result = linePosition(core.toPoint(start))
    } else {
      const s = core.bol(start.line)
      const e = core.eol(start.line)
      if (!s || !e) return undefined
      result = {
        start: s,
        end: e,
      }
    }

    if (end && result) {
      const endLR = linePosition(end)
      if (!endLR) return undefined
      result.end = endLR.end
    }

    return result
  }
  return {
    ...core,
    linePosition,
  }
}
