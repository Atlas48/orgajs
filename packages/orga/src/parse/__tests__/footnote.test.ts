import { parse } from './utils'

it('can handle footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle multi line footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
And here is *another* line.
`.trim()
    )
  ).toMatchSnapshot()
})

it('can handle complex footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
#+BEGIN_SRC javascript
console.log('footnote with code block')
#+END_SRC
`.trim()
    )
  ).toMatchSnapshot()
})

it('knows when headline can stop footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
And here is another line.
* A Headline
`.trim()
    )
  ).toMatchSnapshot()
})

it('knows when footnote can stop footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
And here is another line.
[fn:2] another footnote.
`.trim()
    )
  ).toMatchSnapshot()
})

it('knows when empty lines can stop footnote', () => {
  expect(
    parse(
      `
[fn:1] Content of the footnote.
And here is another line.

still belongs to fn:1


This is not.
`.trim()
    )
  ).toMatchSnapshot()
})
