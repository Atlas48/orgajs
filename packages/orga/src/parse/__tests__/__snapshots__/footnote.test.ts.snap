// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can handle complex footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 2,
              "offset": 32,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
        Object {
          "attributes": Object {},
          "name": "SRC",
          "params": Array [
            "javascript",
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 10,
              "line": 4,
              "offset": 104,
            },
            "start": Object {
              "column": 1,
              "line": 2,
              "offset": 32,
            },
          },
          "type": "block",
          "value": "
console.log('footnote with code block')",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 10,
          "line": 4,
          "offset": 104,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 10,
      "line": 4,
      "offset": 104,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 32,
              "line": 1,
              "offset": 31,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 32,
          "line": 1,
          "offset": 31,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 32,
      "line": 1,
      "offset": 31,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`can handle multi line footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
                "start": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
              },
              "type": "text.plain",
              "value": " ",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 13,
                  "line": 2,
                  "offset": 44,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
              },
              "type": "text.plain",
              "value": "And here is ",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 22,
                  "line": 2,
                  "offset": 53,
                },
                "start": Object {
                  "column": 13,
                  "line": 2,
                  "offset": 44,
                },
              },
              "type": "text.bold",
              "value": "another",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 28,
                  "line": 2,
                  "offset": 59,
                },
                "start": Object {
                  "column": 22,
                  "line": 2,
                  "offset": 53,
                },
              },
              "type": "text.plain",
              "value": " line.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 28,
              "line": 2,
              "offset": 59,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 28,
          "line": 2,
          "offset": 59,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 28,
      "line": 2,
      "offset": 59,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`knows when empty lines can stop footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
                "start": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
              },
              "type": "text.plain",
              "value": " ",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 26,
                  "line": 2,
                  "offset": 57,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
              },
              "type": "text.plain",
              "value": "And here is another line.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 4,
              "offset": 59,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 22,
                  "line": 4,
                  "offset": 80,
                },
                "start": Object {
                  "column": 1,
                  "line": 4,
                  "offset": 59,
                },
              },
              "type": "text.plain",
              "value": "still belongs to fn:1",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 6,
              "offset": 82,
            },
            "start": Object {
              "column": 1,
              "line": 4,
              "offset": 59,
            },
          },
          "type": "paragraph",
        },
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 13,
                  "line": 7,
                  "offset": 95,
                },
                "start": Object {
                  "column": 1,
                  "line": 7,
                  "offset": 83,
                },
              },
              "type": "text.plain",
              "value": "This is not.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 13,
              "line": 7,
              "offset": 95,
            },
            "start": Object {
              "column": 1,
              "line": 7,
              "offset": 83,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 13,
          "line": 7,
          "offset": 95,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 7,
      "offset": 95,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`knows when footnote can stop footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
                "start": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
              },
              "type": "text.plain",
              "value": " ",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 26,
                  "line": 2,
                  "offset": 57,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
              },
              "type": "text.plain",
              "value": "And here is another line.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 3,
              "offset": 58,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 1,
          "line": 3,
          "offset": 58,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 25,
                  "line": 3,
                  "offset": 82,
                },
                "start": Object {
                  "column": 8,
                  "line": 3,
                  "offset": 65,
                },
              },
              "type": "text.plain",
              "value": "another footnote.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 25,
              "line": 3,
              "offset": 82,
            },
            "start": Object {
              "column": 8,
              "line": 3,
              "offset": 65,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "2",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 25,
          "line": 3,
          "offset": 82,
        },
        "start": Object {
          "column": 8,
          "line": 3,
          "offset": 65,
        },
      },
      "type": "footnote",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 25,
      "line": 3,
      "offset": 82,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;

exports[`knows when headline can stop footnote 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "attributes": Object {},
          "children": Array [
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
                "start": Object {
                  "column": 8,
                  "line": 1,
                  "offset": 7,
                },
              },
              "type": "text.plain",
              "value": "Content of the footnote.",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
                "start": Object {
                  "column": 32,
                  "line": 1,
                  "offset": 31,
                },
              },
              "type": "text.plain",
              "value": " ",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 26,
                  "line": 2,
                  "offset": 57,
                },
                "start": Object {
                  "column": 1,
                  "line": 2,
                  "offset": 32,
                },
              },
              "type": "text.plain",
              "value": "And here is another line.",
            },
          ],
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 1,
              "line": 3,
              "offset": 58,
            },
            "start": Object {
              "column": 8,
              "line": 1,
              "offset": 7,
            },
          },
          "type": "paragraph",
        },
      ],
      "label": "1",
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 1,
          "line": 3,
          "offset": 58,
        },
        "start": Object {
          "column": 8,
          "line": 1,
          "offset": 7,
        },
      },
      "type": "footnote",
    },
    Object {
      "children": Array [
        Object {
          "actionable": false,
          "children": Array [
            Object {
              "level": 1,
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 2,
                  "line": 3,
                  "offset": 59,
                },
                "start": Object {
                  "column": 1,
                  "line": 3,
                  "offset": 58,
                },
              },
              "type": "stars",
            },
            Object {
              "parent": [Circular],
              "position": Object {
                "end": Object {
                  "column": 13,
                  "line": 3,
                  "offset": 70,
                },
                "start": Object {
                  "column": 3,
                  "line": 3,
                  "offset": 60,
                },
              },
              "type": "text.plain",
              "value": "A Headline",
            },
          ],
          "content": "A Headline",
          "level": 1,
          "parent": [Circular],
          "position": Object {
            "end": Object {
              "column": 13,
              "line": 3,
              "offset": 70,
            },
            "start": Object {
              "column": 1,
              "line": 3,
              "offset": 58,
            },
          },
          "type": "headline",
        },
      ],
      "level": 1,
      "parent": [Circular],
      "position": Object {
        "end": Object {
          "column": 13,
          "line": 3,
          "offset": 70,
        },
        "start": Object {
          "column": 1,
          "line": 3,
          "offset": 58,
        },
      },
      "properties": Object {},
      "type": "section",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 13,
      "line": 3,
      "offset": 70,
    },
    "start": Object {
      "column": 8,
      "line": 1,
      "offset": 7,
    },
  },
  "properties": Object {},
  "type": "document",
}
`;
